libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -101,12 +101,14 @@ string IRGenerator::generate(ContractDefinition const& _contract)
 	t("memoryInit", memoryInit());
 	t("constructor", constructorCode(_contract));
 	t("deploy", deployCode(_contract));
+	generateConstructorFunctions(_contract);
```

Comment:
Do we really need this? Is the reason because there is not always a FunctionDefinition for the constructor?

Reply:
Yes. And it generates constructors for all contracts in inheritance chain if there is no user defined